ui/src/app/store/manage-collectiom/reducers.ts

```diff
@@ -56,7 +76,139 @@ const reducer: ActionReducer<CollectionEntityState> = createReducer(
   on(loadSubjectsFail, addSubjectFail, editSubjectFail, deleteSubjectFail, deleteSubjectSuccess, state => ({ ...state, isPending: false })),
   on(loadSubjectsSuccess, (state, { subjects }) => ({ ...state, isPending: false, subjects })),
   on(setSelectedSubject, (state, { subject }) => ({ ...state, subject })),
-  on(resetSubjects, () => ({ ...initialState }))
+  on(resetSubjects, () => ({ ...initialState })),
+  on(getSubjectExamples, (state) => ({...state, isCollectionPending: true})),
+  on(getSubjectExamplesSuccess, (state, {items, apiKey}) => {
+	  const collectionCopy = [...state.collection.filter(item => item.status !== CircleLoadingProgressEnum.Uploaded)];
+
+	  const newCollectionItems = items.map(item => ({
+		  url: CollectionHelper.getCollectionItemUrl(apiKey, item.image_id),
+		  id: item.image_id,
+		  status: CircleLoadingProgressEnum.Uploaded,
+		  subject: item.subject
+	  }));
+
+	  return {
+		  ...state,
+		  isCollectionPending: false,
+		  collection:[...newCollectionItems, ...collectionCopy]
+	  };
+  }),
+  on(getSubjectExamplesFail, (state) => ({...state, isCollectionPending: false})),
+  on(addFileToCollection, (state, {url, file, subject}) => ({
+	  ...state,
+	  collection: [
+		  ...state.collection,
+		  {url, file, subject, status: CircleLoadingProgressEnum.OnHold}
+		]
+	})
+  ),
+  on(uploadImage, (state, {item}) => {
+	const collectionCopy = [...state.collection];
+	const targetItemIndex = collectionCopy.findIndex((collectionCopyItem) => item.url === collectionCopyItem.url);
+
+	if (~targetItemIndex) {
+		collectionCopy[targetItemIndex] = {
+			...collectionCopy[targetItemIndex],
+			status: CircleLoadingProgressEnum.InProgress
+		}
+	}
+
+	return ({
+	  ...state,
+	  collection: collectionCopy
+	});
+  }),
+  on(uploadImageSuccess, (state, {item}) => {
+	const collectionCopy = [...state.collection];
+	const targetItemIndex = collectionCopy.findIndex((collectionCopyItem) => item.url === collectionCopyItem.url);
+
+	if (~targetItemIndex) {
+		collectionCopy[targetItemIndex] = {
+			...collectionCopy[targetItemIndex],
+			status: CircleLoadingProgressEnum.Uploaded
+		}
+	}
+
+	return ({
+		...state,
+		collection: collectionCopy
+	});
+  }),
+  on(uploadImageFail, (state, {item, error}) => {
+	const collectionCopy = [...state.collection];
+	const targetItemIndex = collectionCopy.findIndex((collectionCopyItem) => item.url === collectionCopyItem.url);
+
+	if (~targetItemIndex) {
+		collectionCopy[targetItemIndex] = {
+			...collectionCopy[targetItemIndex],
+			status: CircleLoadingProgressEnum.Failed,
+			error
+		}
+	}
+
+	return ({
+		...state,
+		collection: collectionCopy
+	});
+  }),
+  on(deleteSubjectExample, (state, {item}) => {
```

Comment:
same as `uploadImage` reducer - can be combined

Reply:
fixed